File: Modelos Java/MatrizJava/src/main/java/com/mycompany/matrizjava/Controlador/Controlador.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.matrizjava.Controlador;

/**
 *
 * @author 57320
 */
import com.mycompany.matrizjava.Modelo.Modelo;
import com.mycompany.matrizjava.Vista.Vista;
import java.util.Scanner;

public class Controlador {
    private Modelo modelo = new Modelo();
    private Vista vista = new Vista();

    public void ejecutar() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese el número de políticos: ");
        int n = sc.nextInt();

        if (n < 10) {
            System.out.println("El número mínimo es 10. Se asignará n = 10 por defecto.");
            n = 10;
        }

        System.out.println("Seleccione el algoritmo de ordenamiento:");
        System.out.println("1. Bubble Sort");
        System.out.println("2. Selection Sort");
        System.out.print("Opción: ");
        int algoritmo = sc.nextInt();
        if (algoritmo != 1 && algoritmo != 2) {
            System.out.println("Opción inválida. Se usará Bubble Sort por defecto.");
            algoritmo = 1;
        }

        modelo.generarPoliticos(n);
        vista.mostrarMatriz(modelo.matrizPoliticos, "Matriz de políticos generada:");

        modelo.ordenarPorEdad(algoritmo);
        vista.mostrarMatriz(modelo.matrizPoliticos, "Filas ordenadas por edad:");

        modelo.ordenarPorDineroYEdad(algoritmo);
        vista.mostrarMatriz(modelo.matrizPoliticos, "Filas por dinero y columnas por edad:");
    }
}

File: Modelos Python/controlador/controlador.py
from modelo.modelo import Modelo
from vista.vista import Vista

class Controlador:
    def __init__(self):
        self.modelo = Modelo()
        self.vista = Vista()

    def ejecutar(self):
        try:
            n = int(input("Ingrese el numero de politicos (minimo 10): "))
        except:
            n = 10
        if n < 10:
            print("Se asignara n = 10 por defecto.")
            n = 10

        estructura = int(input("Seleccione la estructura de datos:\n1. Array\n2. Matriz\nOpcion: "))
        algoritmo = int(input("Seleccione el algoritmo de ordenamiento:\n1. Bubble Sort\n2. Selection Sort\nOpcion: "))
        if algoritmo not in [1, 2]:
            print("Opcion invalida. Se usara Bubble Sort por defecto.")
            algoritmo = 1

        if estructura == 1:
            self.modelo.generar_array(n)
            self.vista.mostrar_array(self.modelo.arrayPoliticos, "\nArray de politicos generado:")
            self.modelo.ordenar_array(algoritmo)
            self.vista.mostrar_array(self.modelo.arrayPoliticos, "\nArray ordenado por dinero robado:")
        else:
            self.modelo.generar_matriz(n)
            self.vista.mostrar_matriz(self.modelo.matrizPoliticos, "\nMatriz de politicos generada:")
            self.modelo.ordenar_matriz(algoritmo)
            self.vista.mostrar_matriz(self.modelo.matrizPoliticos, "\nMatriz ordenada con Filas por dinero robado y columnas por edad:")
